### ISSUE TITLE
Image stylization example only produces black images when run on CPU

### ISSUE THREAD

**Issue by panmari** [tensorflow/magenta#570]
With tensorflow on master (commit baa85cbf5e51a21f58bc28ef9eedc122e6118eb8). Command run:

    CUDA_VISIBLE_DEVICES=-1 bazel-bin/magenta/models/image_stylization/image_stylization_transform       --num_styles=32       --checkpoint=$HOME/Downloads/multistyle-pastiche-generator-varied.ckpt      --input_image=${INPUT_IMG}       --which_styles="[0]"       --output_dir=${OUTPUT_FOLDER}       --output_basename="stylized"

Everything works as intended when running on GPU with `CUDA_VISIBLE_DEVICES=0`. Tensorflow wheel built with 

    bazel build -c opt --copt=-mavx --copt=-mavx2 --copt=-mfma --copt=-mfpmath=both --copt=-msse4.2 --config=cuda //tensorflow/tools/pip_package:build_pip_package

**Comment by psoulos** [comment]
I am running into this issue as well. To perform image stylization on CPU, it is necessary to install tensorflow 1.0.1 instead of tensorflow 1.1.0

**Comment by adrich** [comment]
@psoulos Thank you so much. Had the same problem today and searched for hours. 
Can anybody explain what causes this or how to support the new tensorflow version?

**Comment by panmari** [comment]
I could narrow it down to a problem with the second convolutional layer - all goes black there when running on the CPU. Here's the snippet from `magenta/models/image_stylization/model.py`: 

```python
with tf.variable_scope('transformer', reuse=reuse):
    with slim.arg_scope(
        [slim.conv2d],
        activation_fn=tf.nn.relu,
        normalizer_fn=normalizer_fn,
        normalizer_params=normalizer_params,
        weights_initializer=tf.random_normal_initializer(0.0, 0.01),
        biases_initializer=tf.constant_initializer(0.0)):
      with tf.variable_scope('contract'):
        h = _conv2d(input_, 9, 1, 32, 'conv1')
        # h is identical on CPU and GPU
        h = _conv2d(h, 3, 2, 64, 'conv2')
        # h is identical all black on CPU, GPU works as intended.
        h = _conv2d(h, 3, 2, 128, 'conv3')
```

@fchollet might know more about what changed from 1.0.1 to 1.1.0 regarding convolutions on CPU. They added some cpu-specific logic to `tensorflow/python/layers/convolutional.py`. 
But the bug might also be related to batch normalization or anything else that happens between the two layers.

**Comment by NMB1331** [comment]
How do I install the right version of TensorFlow?

**Comment by psoulos** [comment]
Run

`pip uninstall tensorflow` to get rid of the original version and then run `pip install tensorflow==1.0.1`

**Comment by plastic-kai** [comment]
Same issue on tensorflow v1.2.1

**Comment by AndreJFBico** [comment]
Same issue with latest tensorflow version.

**Comment by paulperrone** [comment]
Still experiencing this issue with the latest versions of TF and Magenta:

magenta-0.3.7 
tensorflow-1.7.0

I am running on CPU only. Ubuntu 16.04.

**Comment by dozeBoy** [comment]
Got to work by installing:

**Comment by hav4ik** [comment]
The problem is caused by `NaN` values in the weights of `Conv2D` layers. The reason why the network works on GPU is supposedly because GPU does not have a exception mechanism, so `NaN` values will be converted to zeros. On CPU, however, the `NaN` is preserved thorough computation.

My personal workaround of this issue is the following:
```python
for var in tf.global_variables():
    w = var.eval()
    w = np.nan_to_num(w)
    var.assign(w).eval()
```
This snippet should be placed every time you load the model into the session. Worked for `tensorflow==1.8.0`. Will make a pull request soon onto both [tensorflow/magenta](https://github.com/tensorflow/magenta) and [tensorflow/magenta-demos](https://github.com/tensorflow/magenta-demos) repositories.

A more correct solution should be replacing all the weights in the model according to the snippet above (so someone should upload the new weights onto [download.magenta.tensorflow.org/models/](download.magenta.tensorflow.org/models/) upstream).

**Comment by TheodoreGalanos** [comment]
Are there any updates on this? I am also having this error and I'm unable to create any images at all. Tried downgrading to tf 1.0.1 without success. @hav4ik would it be easy to share how you edited the scripts with the above code? Thanks in advance.

**Comment by piggy2303** [comment]
this fuction _multiple_styles in file image_stylization_transform.py just like this



    def _multiple_styles(input_image, which_styles, output_dir):
        with tf.Graph().as_default(), tf.Session() as sess:
            mixture = _style_mixture(which_styles, FLAGS.num_styles)
            stylized_images = model.transform(
                input_image,
                alpha=FLAGS.alpha,
                normalizer_fn=ops.weighted_instance_norm,
                normalizer_params={
                    'weights': tf.constant(mixture),
                    'num_categories': FLAGS.num_styles,
                    'center': True,
                    'scale': True
                })
            _load_checkpoint(sess, FLAGS.checkpoint)
    
            for var in tf.global_variables():
                w = var.eval()
                w = np.nan_to_num(w)
                var.assign(w).eval()
    
            stylized_images = stylized_images.eval()
            stylized_images = stylized_images[0]
            stylized_images = (stylized_images*255).astype(np.uint8)
            img = Image.fromarray(stylized_images)
            img.save("name.jpg")
            return img
